scrambles/src/main/java/org/worldcubeassociation/tnoodle/puzzle/SquareOnePuzzle.java

```diff
@@ -196,32 +200,82 @@ protected int getRandomMoveCount() {
+                    currentMin = moveCost;
+                    bestSlashable = possState;
+                }
+            }
+
+            // absolutely no successor found to make it slashable
+            if (bestSlashable == null) {
+                return null;
+            }
+
+            AlgorithmBuilder ab = new AlgorithmBuilder(this, AlgorithmBuilder.MergingMode.CANONICALIZE_MOVES, sqState);
+
+            try {
+                ab.appendMove(bestSlashable.getKey());
+
+                String nextBestSolution = bestSlashable.getValue().solveIn(n);
+                // initially just hope that it cancels with the n-move optimal solution
+                ab.appendAlgorithm(nextBestSolution);
+
+                if (ab.getTotalCost() > n) {
+                    // didn't cancel, try and find optimal solution in n-1
+                    ab = new AlgorithmBuilder(this, AlgorithmBuilder.MergingMode.CANONICALIZE_MOVES, sqState);
+                    ab.appendMove(bestSlashable.getKey());
+
+                    String oneMoveLessSolution = bestSlashable.getValue().solveIn(n - 1);
+                    // no need for any further checks here.
+                    // the "worst" thing that can happen is the n-1 move solution happens to cancel
+                    // (for whatever reason), in which case we've kept the bounds anyways
+                    ab.appendAlgorithm(oneMoveLessSolution);
+                }
+
+                return ab.getStateAndGenerator().generator;
+            } catch (InvalidMoveException e) {
+                e.printStackTrace();
+            }
+
+            // some invalid move exception happened; normally we don't ever get here
+            return null;
+        }
+
+        FullCube f = sqState.toFullCube();
+        String scramble = twoPhaseSearcher.get().solutionOpt(f, n);
         return scramble == null ? null : scramble.trim();
     }
-    */
 
     static HashMap<String, Integer> costsByMove = new HashMap<String, Integer>();
+    static HashMap<String, Integer> topBottomCostsByMove = new HashMap<String, Integer>();
     static {
         for(int top = -5; top <= 6; top++) {
             for(int bottom = -5; bottom <= 6; bottom++) {
                 if(top == 0 && bottom == 0) {
                     // No use doing nothing =)
                     continue;
                 }
-                //int topCost = top % 12 == 0 ? 0 : 1;
-                //int bottomCost = bottom % 12 == 0 ? 0 : 1;
-                //int cost = topCost + bottomCost;
+                int topCost = Math.abs(top);
+                int bottomCost = Math.abs(bottom);
+                int topBottomCost = topCost + bottomCost;
                 int cost = 1;
                 String turn = "(" + top + "," + bottom + ")";
                 costsByMove.put(turn, cost);
+                topBottomCostsByMove.put(turn, topBottomCost);
             }
         }
         costsByMove.put("/", 1);
```

Comment:
I don't see a similar insertion for `topBottomCostsByMove`, where is the cost of "/" taken into account?
Actually, looking a bit deeper in the code, it seems that [`getMoveCost`](https://github.com/thewca/tnoodle-lib/pull/5/files#diff-9230152573c468cd8acaf707beb730e2L310) still refers to `costsByMove`, which will make the evaluation of `getMoveCost` in [`AlgorithmBuilder`](https://github.com/thewca/tnoodle-lib/blob/8bea5cba937c38f7a890ad96cb51af2e641acfe3/scrambles/src/main/java/org/worldcubeassociation/tnoodle/scrambles/AlgorithmBuilder.java#L168) return the wrong cost!

Pretty sure the code is wrong right now, but I may very well be missing something.
The fix looks straightforward: just rename the variable in this statement and the `getMoveCost` function, and get rid of the map containing the old mapping :)

Reply:
Wrong button, sorry :/

Reply:
We do not need to consider the Slash move for `topBottomCostsByMove`, as it does not alter "slashability" as per my above comment.

Don't worry, the `costsByMove` map which is used throughout the codebase defines the correct metric as per Reg [12c4](https://www.worldcubeassociation.org/regulations/#12c4)!!